apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/page.tsx

```diff
@@ -0,0 +1,28 @@
+import { redirect } from "next/navigation";
+import { getApiKeys } from "../../../../api/lib/getAPIKeys";
+import { getAuthToken } from "../../../../api/lib/getAuthToken";
+import { InAppWalletsPage } from "./InAppWalletsPage";
+
+export default async function Page(props: {
+  searchParams: {
+    clientId?: string;
+  };
+}) {
+  const authToken = getAuthToken();
+  const { clientId } = props.searchParams;
+
+  if (!authToken) {
+    const params = new URLSearchParams();
+    params.append("next", "/dashboard/connect/in-app-wallets");
+    if (clientId) {
+      params.append("clientId", clientId);
+    }
+    redirect(`/login?${params.toString()}`);
+  }
+
+  const apiKeys = await getApiKeys();
+
+  return <InAppWalletsPage defaultClientId={clientId} apiKeys={apiKeys} />;
+}
+
+export const dynamic = "force-dynamic";
```

Comment:
same thing here, do we need this if we're already accessing cookies?

Reply:
removed